src/options.rs

```diff
@@ -118,6 +120,23 @@ impl SftpOptions {
         self.read_end_buffer_size
             .unwrap_or_else(|| NonZeroUsize::new(1024).unwrap())
     }
+
+    /// Set the write buffer limit for tokio compat file.
+    /// If the write buffer size is larger than the limit in tokio compat file,
+    /// then the buffer will be flushed and following async write operation will
+    /// be pending until flush operation is completed.
+    ///
+    /// It is set to usize::MAX by default.
+    #[must_use]
+    pub const fn tokio_compat_file_write_limit(mut self, limit: NonZeroUsize) -> Self {
+        self.tokio_compat_file_write_limit = Some(limit);
+        self
+    }
+
+    pub(super) fn get_tokio_compat_file_write_limit(&self) -> NonZeroUsize {
+        self.tokio_compat_file_write_limit
+            .unwrap_or_else(|| NonZeroUsize::new(usize::MAX).unwrap())
+    }
```

Comment:
Since we store `usize` instead of `NonZeroUsize` in `Auxiliary` anyway, there's no need to construct a `NonZeroUsize` with another `unwrap`:

```suggestion
    pub(super) fn get_tokio_compat_file_write_limit(&self) -> usize {
        self.tokio_compat_file_write_limit
            .map(NonZeroUsize::get)
            .unwrap_or(usize::MAX)
    }
```